diff --git a/toolchain/apple/toolchain.gni b/toolchain/apple/toolchain.gni
index c978827..e1a0503 100644
--- a/toolchain/apple/toolchain.gni
+++ b/toolchain/apple/toolchain.gni
@@ -508,7 +508,7 @@ template("single_apple_toolchain") {
     tool("cc") {
       depfile = "{{output}}.d"
       precompiled_header_type = "gcc"
-      command = "$coverage_wrapper$cc $md -MF $depfile {{defines}} {{include_dirs}} {{cflags}} {{cflags_c}} -fmodule-name=\"{{cc_module_name}}_Private\"  -c {{source}} -o {{output}}"
+      command = "$coverage_wrapper$cc $md -MF $depfile {{defines}} {{include_dirs}} {{cflags}} {{cflags_c}} -c {{source}} -o {{output}}"
       depsformat = "gcc"
       description = "CC {{output}}"
       outputs = [ "$object_subdir/{{source_name_part}}.o" ]
@@ -518,7 +518,7 @@ template("single_apple_toolchain") {
       depfile = "{{output}}.d"
       precompiled_header_type = "gcc"
 
-      command = "$coverage_wrapper$cxx $md -MF $depfile {{defines}} {{include_dirs}} {{cflags}} {{cflags_cc}} {{module_deps}} -fmodule-name=\"{{cc_module_name}}_Private\" -c {{source}} -o {{output}}"
+      command = "$coverage_wrapper$cxx $md -MF $depfile {{defines}} {{include_dirs}} {{cflags}} {{cflags_cc}} -c {{source}} -o {{output}}"
       depsformat = "gcc"
       description = "CXX {{output}}"
       outputs = [ "$object_subdir/{{source_name_part}}.o" ]
@@ -530,7 +530,7 @@ template("single_apple_toolchain") {
 
       # Module file doesn't need coverage instrumentation because module files
       # represent interfaces rather than implementations.
-      command = "$cxx -MD -MF $depfile {{defines}} {{include_dirs}} {{cflags}} {{cflags_cc}} {{module_deps_no_self}} -fmodule-name=\"{{cc_module_name}}\" -c -x c++ -Xclang -emit-module {{source}} -o {{output}}"
+      command = "this-should-never-run"
       depsformat = "gcc"
       description = "CXX_MODULE {{output}}"
       outputs = [ "$object_subdir/{{source_name_part}}.pcm" ]
diff --git a/toolchain/gcc_toolchain.gni b/toolchain/gcc_toolchain.gni
index d8457bf..4d74e6a 100644
--- a/toolchain/gcc_toolchain.gni
+++ b/toolchain/gcc_toolchain.gni
@@ -322,9 +322,6 @@ template("single_gcc_toolchain") {
 
     # -fmodule-name is not supported by gcc.
     module_name_flag = ""
-    if (is_clang) {
-      module_name_flag = " -fmodule-name=\"{{cc_module_name}}_Private\""
-    }
 
     enable_linker_map = defined(invoker.enable_linker_map) &&
                         invoker.enable_linker_map && generate_linker_map
@@ -363,7 +360,7 @@ template("single_gcc_toolchain") {
 
       # {{output}} may contain .dwo file, so use $output instead of {{output}}
       # in command.
-      command = "$coverage_wrapper$cxx $md -MF $depfile {{defines}} {{include_dirs}} {{cflags}} {{cflags_cc}}${extra_cppflags}${extra_cxxflags} {{module_deps}}${module_name_flag} -c {{source}} -o $output"
+      command = "$coverage_wrapper$cxx $md -MF $depfile {{defines}} {{include_dirs}} {{cflags}} {{cflags_cc}}${extra_cppflags}${extra_cxxflags}${module_name_flag} -c {{source}} -o $output"
       depsformat = "gcc"
       description = "CXX {{output}}"
       outputs = [ output ]
@@ -374,7 +371,7 @@ template("single_gcc_toolchain") {
 
       # Module file doesn't need coverage instrumentation because module files
       # represent interfaces rather than implementations.
-      command = "$cxx -MD -MF $depfile {{defines}} {{include_dirs}} {{cflags}} {{cflags_cc}}${extra_cppflags}${extra_cxxflags} {{module_deps_no_self}} -fmodule-name=\"{{cc_module_name}}\" -x c++ -Xclang -emit-module -c {{source}} -o {{output}}"
+      command = "this-should-never-run"
       depsformat = "gcc"
       description = "CXX_MODULE {{output}}"
       outputs = [ "$object_subdir/{{source_name_part}}.pcm" ]
diff --git a/toolchain/win/toolchain.gni b/toolchain/win/toolchain.gni
index 5a8e519..6f1694f 100644
--- a/toolchain/win/toolchain.gni
+++ b/toolchain/win/toolchain.gni
@@ -220,9 +220,6 @@ template("msvc_toolchain") {
 
     # -fmodule-name is not supported by MSVC.
     module_name_flag = ""
-    if (toolchain_is_clang) {
-      module_name_flag = " \"-fmodule-name={{cc_module_name}}_Private\""
-    }
 
     tool("cc") {
       precompiled_header_type = "msvc"
@@ -245,7 +242,7 @@ template("msvc_toolchain") {
       depsformat = "msvc"
       description = "CXX_MODULE {{output}}"
       outputs = [ "$object_subdir/{{source_name_part}}.pcm" ]
-      command = "$env_wrapper$cl /Fo{{output}} /nologo $show_includes $sys_include_flags{{defines}} {{include_dirs}} {{cflags}} {{cflags_cc}} {{module_deps_no_self}} -fmodule-name=\"{{cc_module_name}}\" /Fd\"$pdbname\" /c -x c++ -Xclang -emit-module {{source}}"
+      command = "this-should-never-run"
     }
 
     tool("cxx") {
@@ -261,7 +258,7 @@ template("msvc_toolchain") {
 
       # Note that the code coverage wrapper scripts assumes that {{source}}
       # comes immediately after /c.
-      command = "$coverage_wrapper$env_wrapper$cl /c {{source}} /Fo{{output}} /nologo $show_includes $sys_include_flags{{defines}} {{include_dirs}} {{cflags}} {{cflags_cc}} /TP {{module_deps}}${module_name_flag} /Fd\"$pdbname\""
+      command = "$coverage_wrapper$env_wrapper$cl /c {{source}} /Fo{{output}} /nologo $show_includes $sys_include_flags{{defines}} {{include_dirs}} {{cflags}} {{cflags_cc}} /TP ${module_name_flag} /Fd\"$pdbname\""
     }
 
     tool("rc") {
